src/OpenLoco/src/Windows/MapWindow.cpp

```diff
@@ -216,12 +223,672 @@ namespace OpenLoco::Ui::Windows::MapWindow
+                mapAltPtr += kRenderedMapWidth + 1;
+                continue;
+            }
+
+            PaletteIndex_t colourFlash0{}, colourFlash1{}, colour0{}, colour1{};
+            auto tile = TileManager::get(pos);
+            for (auto& el : tile)
+            {
+                switch (el.type())
+                {
+                    case ElementType::surface:
+                    {
+                        auto* surfaceEl = el.as<SurfaceElement>();
+                        if (surfaceEl == nullptr)
+                            continue;
+
+                        if (surfaceEl->water() == 0)
+                        {
+                            const auto* landObj = ObjectManager::get<LandObject>(surfaceEl->terrain());
+                            const auto* landImage = Gfx::getG1Element(landObj->mapPixelImage);
+                            colourFlash0 = colourFlash1 = landImage->offset[0];
+                        }
+                        else
+                        {
+                            const auto* waterObj = ObjectManager::get<WaterObject>();
+                            const auto* waterImage = Gfx::getG1Element(waterObj->mapPixelImage);
+                            colourFlash0 = colourFlash1 = waterImage->offset[0];
+                        }
+
+                        colour0 = colour1 = colourFlash0;
+                        break;
+                    }
+
+                    case ElementType::building:
+                    case ElementType::industry:
+                        // Vanilla omits the ghost check, but I think it should be here anyway.
+                        if (!el.isGhost())
+                        {
+                            colour0 = colourFlash0 = PaletteIndex::index_0B;
+                            colourFlash1 = colourFlash0;
+                            colour1 = colour0;
+                        }
+                        break;
+
+                    case ElementType::track:
+                    case ElementType::station:
+                    case ElementType::road:
+                    {
+                        if (el.isGhost() || el.isAiAllocated())
+                            continue;
+
+                        auto owner = CompanyId::null;
+
+                        if (auto* stationEl = el.as<StationElement>())
+                        {
+                            auto station = StationManager::get(stationEl->stationId());
+                            owner = station->owner;
+                        }
+                        else if (auto* trackEl = el.as<TrackElement>())
+                        {
+                            owner = trackEl->owner();
+                        }
+                        else if (auto* roadEl = el.as<RoadElement>())
+                        {
+                            owner = roadEl->owner();
+                        }
+
+                        auto companyColour = CompanyManager::getCompanyColour(owner);
```

Comment:
if the owner == 0xF i.e. the neutral owner (for roads) then it should do colour 0xB like industry and building.